shared/common-adapters/oriented-image.desktop.js

```diff
@@ -51,43 +46,39 @@ class OrientedImage extends React.Component<Props, State> {
   }
 
   state = {
-    cacheStyleTransforms: {},
-    styleTransform: {transform: ''},
+    styleTransform: '',
   }
 
+  _cacheStyleTransforms = {}
   _hasComponentMounted = false
 
+  _handleData = img => {
+    const orientation: number = EXIF.getTag(img, 'Orientation')
+    // EXIF.getTag can return undefined for Orientation if the field is not found
+    if (isNumber(orientation)) {
+      const newTransform: string = makeStyleTransform(orientation)
+      this.setState(p => (p.styleTransform === newTransform ? undefined : {styleTransform: newTransform}))
+    }
+  }
+
   _setTranformForExifOrientation(src) {
-    if (this._hasComponentMounted) {
-      if (this.state.cacheStyleTransforms[src]) {
-        return this.setState({
-          styleTransform: this.state.cacheStyleTransforms[src],
-        })
-      }
+    if (!this._hasComponentMounted) {
+      return
+    }
 
-      const component = this
-      // EXIF will make an HTTP request locally to 127.0.0.1:* to fetch the
-      // image that the keybase service is serving
-      // img = this refers to the image ArrayBuffer fetched from the local server.
-      EXIF.getData({src}, function() {
-        const img = this
-        const orientation: number = EXIF.getTag(img, 'Orientation')
-        // EXIF.getTag can return undefined for Orientation if the field is not found
-        if (isNumber(orientation)) {
-          const styleTransform: Transform = makeStyleTransform(orientation)
-          component.setState(prevState => {
-            const newTransform =
-              prevState.styleTransform.transform !== styleTransform.transform
-                ? styleTransform
-                : prevState.styleTransform
-            return {
-              cacheStyleTransforms: {[src]: newTransform},
-              styleTransform: newTransform,
-            }
-          })
-        }
+    if (this._cacheStyleTransforms[src]) {
```

Comment:
if you have this you can skip all this and just initialize your state with it in your constructor